Sources/GRPC/CallOptions.swift

```diff
@@ -160,14 +174,22 @@ extension CallOptions {
     }
 
     /// Provide a factory to generate request IDs.
+    #if swift(>=5.6)
+    public static func generated(
+      _ requestIDFactory: @escaping @Sendable () -> String
```

Comment:
Should we promote the `StringFactory` typealias to public here to avoid the extra `#if`?